### PR TITLE
PR should pull config from default branch, master -> main

### LINKED ISSUE #111: baldrick pulls config from fork's master branch
```
File ".../flask/_compat.py", line 39, in reraise
  raise value
        └ AssertionError(b'{"message":"No commit found for the ref master","documentation_url":"https://docs.github.com/v3/repos/conten...
...
File ".../baldrick/github/github_api.py", line 282, in get_file_contents
  return super().get_file_contents(path_to_file, branch=branch)
                                    │                    └ 'master'
                                    └ 'pyproject.toml'
File ".../baldrick/github/github_api.py", line 89, in get_file_contents
  assert response.ok, response.content
         │        │   │        └ <property object at 0x7f51b612ce50>
         │        │   └ <Response [404]>
         │        └ <property object at 0x7f51b612cc70>
         └ <Response [404]>
```

The above log, combined with what I saw in astropy/astropy#11481 

```
baldrick.github.github_api:get_repo_config:144 - Got this combined config from
nstarman/astropy@master:
```

suggests that `baldrick` is attempting to read the config from a fork's `master`, not `upstream`.

This would explain why it stopped running for my PRs because I deleted my fork's `master` branch a long time ago, as suggested in https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#deleting-your-master-branch .

And I clearly remember that the bot used to work despite my fork not having a `master` branch before but I don't remember when it stopped working. I just know it's been that way (broken) for quite some time; just didn't know why until now.

### PR DESCRIPTION
**Pull request by pllim**
Have PR pull config from upstream default branch. Fix #111 

Change default branch in github API from master to main.

# TODO

- [ ] Fix tests. I don't understand why the mock magic isn't seeing the config no more. Things seem to work when I live-test it locally without mock.
- [x] Insert PR number in change log.

### CODE REVIEW

File: baldrick/blueprints/github.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def github_webhook():
         installation = payload['installation']['id']
 
     repo_name = payload['repository']['full_name']
-    repo = RepoHandler(repo_name, installation=installation)
+    repo = RepoHandler(repo_name, branch='main', installation=installation)
```
Comment (pllim) [inline baldrick/blueprints/github.py:43]: What is the blueprint used for anyway? Will this break things?

File: baldrick/plugins/github_towncrier_changelog.py [hunk 2]
```diff
@@ -99,8 +98,7 @@ def verify_pr_number(pr_number, matching_file):
 
 
 def load_towncrier_config(pr_handler):
-    file_content = pr_handler.get_file_contents("pyproject.toml", branch=pr_handler.base_branch)
-    config = loads(file_content)
+    config = pr_handler.get_repo_config()
```
Comment (pllim) [inline baldrick/plugins/github_towncrier_changelog.py:101]: Changing from `get_file_contents` to `get_repo_config` seems to break a lot of tests. I don't understand why -- is it some subtlety in how TOML works or I missed something in the mock magic?

File: baldrick/plugins/tests/test_github_towncrier_changelog.py [hunk 3]
```diff
@@ -14,8 +15,8 @@
     filename = "NEWS.rst"
 
 [ tool.testbot ]
-[ tool.testbot.towncrier_changelog ]
-enabled=true
+  [[ tool.testbot.towncrier_changelog ]]
```
Comment (pllim) [inline baldrick/plugins/tests/test_github_towncrier_changelog.py:18]: Didn't make any difference but I thought this is more consistent with what is in `astropy`.

File: baldrick/github/github_api.py [hunk 4]
```diff
@@ -777,9 +797,10 @@ def get_repo_config(self, branch=None, path_to_file='pyproject.toml'):
             Configuration parameters.
 
         """
+        upstream = RepoHandler(self.base_repo_name)
         if not branch:
-            branch = self.base_branch
-        return super().get_repo_config(branch=branch, path_to_file=path_to_file)
+            branch = upstream.default_branch
```
Comment (Cadair) [inline baldrick/github/github_api.py:802]: Aside from my comments in #111 this change will probably fix the bug (although I don't know why it's needed). We should however, still be using base branch; we always want to pull the config from the branch the PR is targeting, to allow varying configs over different branches.

```suggestion
            branch = upstream.base_branch
```

**Comment by pllim** [comment]
Since you cannot reproduce the bug on your side, I am now not convinced that I went down the correct rabbit hole. Let's debug it a bit more.

**Pull request closed without merging**